File: src/main/java/com/venus/meetspace/dto/request/ActivityUpdateRequest.java
package com.venus.meetspace.dto.request;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 用于传入更新活动的请求体<br>
 * 所有字段可为空
 */
@Data
public class ActivityUpdateRequest {
    private String title;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private LocalDateTime signupDeadline;
    private String address;
    private String image;
    private String description;
}

File: frontend/lib/model/request/activity_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'activity_update_request.g.dart';

@JsonSerializable()
class ActivityUpdateRequest {
  final String? title;
  final DateTime? startTime;
  final DateTime? endTime;
  final DateTime? signupDeadline;
  final String? address;
  final int? minParticipants;
  final int? maxParticipants;
  final String? image;
  final String? description;
  @JsonKey(name: 'latitude')
  final double? latitude;
  @JsonKey(name: 'longitude')
  final double? longitude;

  ActivityUpdateRequest({
    this.title,
    this.startTime,
    this.endTime,
    this.signupDeadline,
    this.address,
    this.minParticipants,
    this.maxParticipants,
    this.image,
    this.description,
    this.latitude,
    this.longitude,
  });

  factory ActivityUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$ActivityUpdateRequestFromJson(json);
  Map<String, dynamic> toJson() => _$ActivityUpdateRequestToJson(this);
}
